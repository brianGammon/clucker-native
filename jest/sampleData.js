const sampleData1 = {
  userSettings: {
    key: 'user1',
    data: {
      currentFlockId: 'flock1',
    },
  },
  flocks: {
    data: {
      flock1: {
        name: 'Test Flock 1',
        ownedBy: 'user1',
      },
      flock2: {
        name: 'Test Flock 2',
        ownedBy: 'user2',
      },
    },
  },
  chickens: {
    data: {
      chicken1: {
        name: 'The First Chicken',
      },
      chicken2: {
        name: 'The Second Chicken',
      },
    },
  },
  eggs: {
    data: {
      egg1: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-01',
        modified: '2018-01-12T18:05:54.806Z',
        notes:
          'This was the record setting egg from May 2017, real egg was 66.1g on this day',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg2: {
        chickenId: 'chicken2',
        chickenName: 'Baby',
        damaged: false,
        date: '2017-09-24',
        modified: '2017-10-12T18:05:54.806Z',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg3: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-02',
        modified: '2018-08-02T18:05:54.806Z',
        notes: '',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 68.1,
      },
    },
  },
};

export default sampleData1;
