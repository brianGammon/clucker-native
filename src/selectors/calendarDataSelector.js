/* @flow */
import Moment from 'moment';
import * as momentRange from 'moment-range';
import createCachedSelector from 're-reselect';
import flockStatsSelector from './flockStatsSelector';

const moment = momentRange.extendMoment(Moment);
const getMonth = (state, month) => month;

const calendarDataSelector = createCachedSelector(
  [flockStatsSelector, getMonth],
  (stats, month) => {
    const date = moment(month);
    const range = moment.range(
      date.clone().startOf('month'),
      date.clone().endOf('month'),
    );

    range.start.subtract(range.start.weekday(), 'days');
    range.end.add(6 - range.end.weekday(), 'days');

    const result = {};
    Array.from(range.by('days')).forEach((day) => {
      const formattedDate = day.format('YYYY-MM-DD');
      result[formattedDate] = (stats && stats.eggsPerPeriod[formattedDate]) || {};
    });

    return result;
  },
)(getMonth);

export default calendarDataSelector;
