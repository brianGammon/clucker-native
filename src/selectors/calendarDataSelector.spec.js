import calendarDataSelector from './calendarDataSelector';
import sampleData from '../../jest/test-data/full.2017-10-to-2018-05.json';

test('calendar range tests', () => {
  const month = '2018-04';
  const calendarData = calendarDataSelector(sampleData.eggs.data, month);
  expect(calendarData).toMatchSnapshot();
});
