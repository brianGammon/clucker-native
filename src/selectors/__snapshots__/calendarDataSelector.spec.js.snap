// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calendar range tests 1`] = `
Object {
  "2018-04-01": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken3": 1,
    },
    "total": 3,
  },
  "2018-04-02": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken4": 1,
    },
    "total": 2,
  },
  "2018-04-03": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken3": 1,
      "chicken4": 1,
    },
    "total": 4,
  },
  "2018-04-04": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
    },
    "total": 2,
  },
  "2018-04-05": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken4": 1,
    },
    "total": 3,
  },
  "2018-04-06": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken4": 1,
    },
    "total": 2,
  },
  "2018-04-07": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken3": 1,
      "chicken4": 1,
    },
    "total": 4,
  },
  "2018-04-08": Object {
    "byChicken": Object {
      "chicken2": 1,
      "chicken4": 1,
    },
    "total": 2,
  },
  "2018-04-09": Object {
    "byChicken": Object {
      "chicken2": 1,
      "chicken4": 1,
    },
    "total": 2,
  },
  "2018-04-10": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken3": 1,
      "chicken4": 1,
    },
    "total": 3,
  },
  "2018-04-11": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken3": 1,
      "chicken4": 1,
    },
    "total": 4,
  },
  "2018-04-12": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
    },
    "total": 2,
  },
  "2018-04-13": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken3": 1,
      "chicken4": 1,
    },
    "total": 3,
  },
  "2018-04-14": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken3": 1,
      "chicken4": 1,
    },
    "total": 4,
  },
  "2018-04-15": Object {
    "byChicken": Object {
      "chicken1": 1,
    },
    "total": 1,
  },
  "2018-04-16": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken3": 1,
      "chicken4": 1,
    },
    "total": 3,
  },
  "2018-04-17": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken3": 1,
      "chicken4": 1,
    },
    "total": 4,
  },
  "2018-04-18": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken4": 1,
    },
    "total": 3,
  },
  "2018-04-19": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
    },
    "total": 2,
  },
  "2018-04-20": Object {
    "byChicken": Object {
      "chicken2": 1,
      "chicken3": 1,
      "chicken4": 1,
    },
    "total": 3,
  },
  "2018-04-21": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken4": 1,
    },
    "total": 3,
  },
  "2018-04-22": Object {
    "byChicken": Object {
      "chicken1": 1,
    },
    "total": 1,
  },
  "2018-04-23": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken4": 1,
    },
    "total": 3,
  },
  "2018-04-24": Object {
    "byChicken": Object {
      "chicken2": 1,
      "chicken4": 1,
    },
    "total": 2,
  },
  "2018-04-25": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken3": 1,
    },
    "total": 3,
  },
  "2018-04-26": Object {
    "byChicken": Object {
      "chicken4": 1,
    },
    "total": 1,
  },
  "2018-04-27": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken4": 1,
    },
    "total": 3,
  },
  "2018-04-28": Object {
    "byChicken": Object {
      "chicken1": 1,
    },
    "total": 1,
  },
  "2018-04-29": Object {
    "byChicken": Object {
      "chicken4": 1,
    },
    "total": 1,
  },
  "2018-04-30": Object {
    "byChicken": Object {
      "chicken1": 1,
      "chicken2": 1,
      "chicken4": 1,
    },
    "total": 3,
  },
  "2018-05-01": Object {},
  "2018-05-02": Object {},
  "2018-05-03": Object {},
  "2018-05-04": Object {},
  "2018-05-05": Object {},
}
`;
